### ISSUE TITLE
Updates on this project

### ISSUE THREAD

**Issue by quantissimo** [glosswordteam/Glossword#10]
Hi Dmitry,

Do you have any plans to revive this project?

**Comment by glosswordteam** [comment]
Yes, I have. I tried php 7, some fixes needed for MySQL8

**Comment by quantissimo** [comment]
Good news. I'd like to see it resurrected. It's one of its kind. Any timeframe?

**Comment by zdenop** [comment]
Any updates for php 7.2?

**Comment by pod-o-mart** [comment]
Great! I use Glossword a lot and I am looking forward to a new release!

**Comment by KyngJohannes** [comment]
Is there any update coming soon? I applied the fix for PHP 7, but the installer has reset after a GoDaddy change, but I can’t reinstall this version without reverting to an old PHP version first.

**Comment by zdenop** [comment]
@KyngJohannes : I managed  to make changes so it runs on php7.4 see https://github.com/sk-spell/Glossword
There are still some problem with editing existing terms (adding new term work fine), but at least it runs...

**Comment by KyngJohannes** [comment]
> @KyngJohannes : I managed to make changes so it runs on php7.4 see https://github.com/sk-spell/Glossword
> There are still some problem with editing existing terms (adding new term work fine), but at least it runs...

Is it possible to install this version directly or is it only for live versions? 

EDIT: Sorry, I just noticed the readme suggests the latter. So I suppose the only solution is to revert to PHP 5.4, reinstall the original, update the files, go back to PHP 7?

**Comment by zdenop** [comment]
@KyngJohannes: I did not try to do make clean install. I just continue with @pod-o-mart changes. Maybe you can try it if work or not.  If not than you need to follow steps you mentioned. BTW: Do you have copy of db or do you plan to create new dictionary?

**Comment by KyngJohannes** [comment]
> @KyngJohannes: I did not try to do make clean install. I just continue with @pod-o-mart changes. Maybe you can try it if work or not. If not than you need to follow steps you mentioned. BTW: Do you have copy of db or do you plan to create new dictionary?

Thank you for your work on this. I have everything exactly as it was on my server, so I’d like to use the same database; I’m just locked in the install script, so I have to reinstall.

**Comment by KyngJohannes** [comment]
I solved an installation issue by renaming the htaccess file, but I can't install because of these failing:

PHP setting mbstring.http_input.
PHP setting mbstring.http_output.

EDIT: Edited the htaccess file to pass them.

**Comment by pod-o-mart** [comment]
Hi,
I do also really need Glossword in a PHP7 or, even better, in a PHP8 version for my [dictionary project](https://ordbog.oesteuropastudier.dk/). I couldn't find any good alternatives to Glossword and I have always been quite happy with it.

I would also be ready to put cash on the table for an upgraded version, so I propose that we collect some money and find someone who could do that. My contribution would be up to 100 €.

**Comment by glosswordteam** [comment]
Sounds challenging )

**Comment by glosswordteam** [comment]
Ok, I can update status on 25th October.

**Comment by pod-o-mart** [comment]
Ok, that sounds good! I will send you the 100 € for sure given the case that the upgrade will come in the nearer future.
Anyways, I hereby ask the other users also to contribute because I think it is worth it. There is no better free dictionary script available anywhere.
I would also have some (not so many) feature requests. I will send them later...

**Comment by pod-o-mart** [comment]
Hi, here my feature requests:

For me personal it would be a great relief, if the visual themes also could be created in files instead of being saved in the database. It is much easier to edit a template by saving a file than by editing a textarea and to save it then. So first my very biggest request:

- Visual themes: Option to create themes in files and not from the database
- Interface language as GET, e.g.
  - www.domain.com/?lang=en or even better like this
  - www.domain.com/en/dictionary/term
- Sitemap: Google, Yandex etc. compatible XML sitemap for dictionary index pages, cutom pages and the individual term pages  (This is a less important request, though)
- Sort order: Merge A/Ä, O/Ö, U/Ü, ß/ss. German does not discriminate between these letters, e.g. "A" and "Ä" is treated as the same letter, as well as "ss" and "ß"  (less important)

**Comment by glosswordteam** [comment]
First, answering questions:

- The storage method for a visual themes can be switchable after a minor update, but import/export procedure needs update also.
- Try index.php?il=en
- Sitemap generation - yes, it is already in wishlist. However sitemap is possible for dictionary index pages and cutom pages only. http://glossword.biz/kb-1p
- As for a sorting order - have you tried a custom sorting order feature?

**Comment by glosswordteam** [comment]
I reviewed the source code completely and what I think:
- Compatibility with the latest PHP version can be solved in two steps: first is for PHP 7.4, next is for PHP 8.
- There are too many differences in PHP 8 to solve them all in a weekend.
- HTML-templates engine needs a major update (originally developed in 2001!) or replacement with an alternative.

Major steps to do:
- A new database driver for mysqli (almost done, testing on installation procedure)
- Replace each() to foreach(), `define` to `const`, create_function() to alternative.

Next update on 1st November.

**Comment by glosswordteam** [comment]
Compatibility with PHP 7.4 is close to be released.

- Database driver for mysqli - done
- A lot of code replacements - done
- Installation procedure - in progress

Next update on 8th November.

**Comment by KyngJohannes** [comment]
Is there a fix for special characters in the pipeline? It may be only ð that is affected.

**Comment by pod-o-mart** [comment]
Hi Glossword Team,
thank you for your effort!
>     * The storage method for a visual themes can be switchable after a minor update, but import/export procedure needs update also.
I don't really understand, what import/export procedure do you mean? Where would the theme files be then? 
>     * Try index.php?il=en
This is a new feature, right? Because this doesn't work in my installation of 1.8.12.
The 2 features above would really save me a lot of work!
>     * Sitemap generation - yes, it is already in wishlist. However sitemap is possible for dictionary index pages and cutom pages only. http://glossword.biz/kb-1p
great!
>     * As for a sorting order - have you tried a custom sorting order feature?
Yes, I use custom sort orders. Anyways, I solved the problem by changing the collation to `utf8_general_ci` in the database

**Comment by glosswordteam** [comment]
>I don't really understand, what import/export procedure do you mean?

System settings -> Maintenance -> Backup is export to xml-file and Visual themes -> Import is import from xml-file

**Comment by glosswordteam** [comment]
>This is a new feature, right?

No, this is old feature, all visual themes provide the selection of interface languages. 
It is possible to change `li` to `lang` in `/inc/constants.inc.php`
```define('GW_LANG_I', 'il');```

**Comment by pod-o-mart** [comment]
> System settings -> Maintenance -> Backup is export to xml-file and Visual themes -> Import is import from xml-file

What I mean is not the backup function. I would like to be able to edit the visual theme not in
`Control panel > Visual themes > "Brand 1.8.12" > edit`
and then e.g.
```
tpl_header
tpl_index
tpl_footer
```
but rather directly by editing e.g.
```
/templates/gw_brand/tpl_header.php
/templates/gw_brand/tpl_index.php
/templates/gw_brand/tpl_footer.php
```
or in some other folder, so the theme wouldn't be saved in the database but in the file system of the Glossword installation. Wouldn't there be some possibility for that?

**Comment by glosswordteam** [comment]
> Wouldn't there be some possibility for that?

Yes, it is possible. However, note that in this case, the Installation, Backup and Import procedures should know how to operate with your files as well.

**Comment by glosswordteam** [comment]
Meanwhile, I started to commit changes to repository.
I met incompatibility issues while testing, so working on it.
Next update on 15th November.

**Comment by glosswordteam** [comment]
Not much say this time - busy holidays.
Also searching for a template engine/processor.

**Comment by indrakurnia** [comment]
hi @glosswordteam , is there any update?
Thanks

**Comment by glosswordteam** [comment]
> hi @glosswordteam , is there any update?
> Thanks

I am on vacation now, away from computer.
No major updates yet, looking forward to that myself.

**Comment by zdenop** [comment]
Hello all,

I must to switch to php 8 because of my hosting. 
I started to make [some changes](https://github.com/sk-spell/Glossword) (e.g. replacing each() ), but I am not php programmer.
At this stage I am not interested at install part or editing themes
At the moment modified code run on php7 and php 8.0 (maybe also 8.2). Login does now work, and php8 indicated there is probably a memory leak....
So it would be great if a real php programmer would start to be involved the code upgrade..
